hadoop-yarn-project/hadoop-yarn/hadoop-yarn-api/pom.xml

```diff
@@ -118,45 +118,21 @@
     </resources>
     <plugins>
       <plugin>
-        <groupId>org.apache.hadoop</groupId>
-        <artifactId>hadoop-maven-plugins</artifactId>
-        <executions>
+        <groupId>org.xolstice.maven.plugins</groupId>
+	<artifactId>protobuf-maven-plugin</artifactId>
+	<executions>
           <execution>
-            <id>compile-protoc</id>
-            <goals>
-              <goal>protoc</goal>
-            </goals>
+            <id>src-compile-protoc</id>
             <configuration>
-              <protocVersion>${protobuf.version}</protocVersion>
-              <protocCommand>${protoc.path}</protocCommand>
-              <imports>
-                <param>${basedir}/../../../hadoop-common-project/hadoop-common/src/main/proto</param>
-                <param>${basedir}/src/main/proto</param>
-                <param>${basedir}/src/main/proto/server</param>
-              </imports>
-              <source>
-                <directory>${basedir}/src/main/proto</directory>
-                <includes>
-                  <include>yarn_protos.proto</include>
-                  <include>yarn_service_protos.proto</include>
-                  <include>applicationmaster_protocol.proto</include>
-                  <include>applicationclient_protocol.proto</include>
-                  <include>containermanagement_protocol.proto</include>
-                  <include>server/yarn_server_resourcemanager_service_protos.proto</include>
-                  <include>server/resourcemanager_administration_protocol.proto</include>
-                  <include>application_history_client.proto</include>
-                  <include>server/application_history_server.proto</include>
-                  <include>client_SCM_protocol.proto</include>
-                  <include>server/SCM_Admin_protocol.proto</include>
-                  <include>yarn_csi_adaptor.proto</include>
-                  <include>YarnCsiAdaptor.proto</include>
-                </includes>
-              </source>
+              <additionalProtoPathElements>
+                <additionalProtoPathElement>
+                  ${basedir}/../../../hadoop-common-project/hadoop-common/src/main/proto
```

Comment:
`${basedir}/src/main/proto/server` missed from additional paths.

Reply:
I think this is not required? The plugin will scan the directory recursively.

Reply:
I don't think so. It just adds them as arguments to protoc command.
Just checks that it should be a directory. No extra processing done on that.

https://github.com/xolstice/protobuf-maven-plugin/blob/master/src/main/java/org/xolstice/maven/plugin/protobuf/AbstractProtocMojo.java
https://github.com/xolstice/protobuf-maven-plugin/blob/master/src/main/java/org/xolstice/maven/plugin/protobuf/Protoc.java

Reply:
I mean it will compile the proto files under the server directory as we set the protoc path to ${basedir}/src/main/proto, which is the parent directory of ${basedir}/src/main/proto/server, so we do not need to add extra 'additionalProtoPathElement' any more.

Reply:
Oh okay.  Got it. Makes sense.
`${basedir}/src/main/proto/server` is also having source protos. Based on previous \<import\> section I was thinking it as just a dependency.